### ISSUE TITLE
Extending Flake8Async rules to nested calls?

### ISSUE THREAD

**Issue by tasercake** [astral-sh/ruff#7807]
Hi there!

I'm interested in adding a rule to identify blocking calls in async contexts. Being new to ruff, I'm hoping to get a sense of how big of a task this would be.

Would it be feasible to have ruff identify _indirect_ calls to blocking functions like `open()` or `requests.get()`?

Here's some examples of the cases I'd like to catch:

```python
import asyncio
import requests

async def foo():
    loop = asyncio.get_running_loop()
    return await loop.run_in_executor(None, requests.get, "https://example.com")  # OK (but AFAIK this isn't explicitly checked by ruff now)

async def bar():
    return requests.get("https://example.com")  # Error! (this is already caught by ASYNC100)

def baz():
    return requests.get("https://example.com")  # OK

def qux():
    return  # OK

async def main():
    await foo()  # OK
    await bar()  # OK
    baz()  # Error! (but this is not caught as of now)
    qux()  # OK
```

The current output looks like this:

```shellsession
$ ruff check . --select ASYNC
main.py:9:12: ASYNC100 Async functions should not call blocking HTTP methods
Found 1 error.
```

**Comment by tasercake** [comment]
Hey Charlie, thanks for the info!

I didn't realise ruff doesn't support cross-file analysis at the moment. My own use case involves a large codebase where most blocking functions are in different files, so I don't think this rule would add much value.

If and when cross-file analysis gets a concrete timeline I'd love to revisit this, but for now I'll explore other solutions 🙂

Please feel free to close this issue

**Comment by charliermarsh** [comment]
Sounds good -- we do plan to support this eventually since it's an unfortunate limitation, but we have a few other priorities ahead of it, since we're able to get pretty far with single-file analysis as is. I've tagged this with a label to revisit when we support multi-file analysis.
